src/azure-cli/azure/cli/command_modules/dms/scenario_inputs.py

```diff
@@ -140,12 +136,12 @@ def get_migrate_mysql_to_azuredbformysql_input(database_options_json,
         if not isinstance(migration_level_settings, dict):
             raise ValidationError('migration_level_settings should be a dictionary')
 
-    if requires_consistent_snapshot:
-        migration_level_settings['enableConsistentBackup'] = 'true'
-    elif has_consistent_snapshot_options:
+    if has_consistent_snapshot_options:
         make_source_server_read_only = database_options_json.get('make_source_server_read_only', False)
         set_optional(migration_level_settings, 'enableConsistentBackup', database_options_json,
                      'enable_consistent_backup')
+        set_optional(migration_level_settings, 'enableConsistentBackupWithoutLocks', database_options_json,
+                     'enable_consistent_backup_without_locks')
```

Comment:
Are these options considered mutually exclusive?

Reply:
These two properties influence one another so they're not mutually exclusive. Ideally this should be an enum of somekind, but that would be a breaking change.